src/Sarif.Multitool/Rules/SarifValidationSkimmerBase.cs

```diff
@@ -59,42 +56,53 @@ protected void LogResult(string jPointer, string formatId, params string[] args)
                 RuleUtilities.BuildResult(DefaultLevel, Context, region, formatId, argsWithPointer));
         }
 
-        protected virtual void Analyze(Attachment attachment, string attachmentPointer)
+        protected virtual void Analyze(Address address, string addressPointer)
         {
         }
 
-        protected virtual void Analyze(CodeFlow codeFlow, string codeFlowPointer)
+        protected virtual void Analyze(Artifact artifact, string artifactPointer)
         {
         }
 
-        protected virtual void Analyze(Conversion conversion, string conversionPointer)
+        protected virtual void Analyze(ArtifactChange artifactChange, string artifactChangePointer)
         {
         }
 
-        protected virtual void Analyze(Edge edge, string edgePointer)
+        protected virtual void Analyze(ArtifactLocation artifactLocation, string artifactLocationPointer)
         {
         }
 
-        protected virtual void Analyze(EdgeTraversal edgeTraversal, string edgeTraversalPointer)
+        protected virtual void Analyze(Attachment attachment, string attachmentPointer)
+        {
+        }
+
+        protected virtual void Analyze(CodeFlow codeFlow, string codeFlowPointer)
+        {
+        }
+        protected virtual void Analyze(ConfigurationOverride configurationOverride, string configurationOverridePointer)
         {
         }
 
-        protected virtual void Analyze(ArtifactChange fileChange, string fileChangePointer)
+        protected virtual void Analyze(Conversion conversion, string conversionPointer)
         {
         }
 
-        protected virtual void Analyze(ArtifactLocation fileLocation, string fileLocationPointer)
+        protected virtual void Analyze(Edge edge, string edgePointer)
         {
         }
 
-        protected virtual void Analyze(Artifact fileData, string filePointer)
+        protected virtual void Analyze(EdgeTraversal edgeTraversal, string edgeTraversalPointer)
         {
         }
 
         protected virtual void Analyze(Graph graph, string graphPointer)
         {
         }
 
+        protected virtual void Analyze(GraphTraversal graphTraversal, string graphTraversalPointer)
+        {
+        }
+
         protected virtual void Analyze(Invocation invocation, string invocationPointer)
```

Comment:
>Analyze [](start = 31, length = 7)

If we were ninja, there would be a test that reflects over this type and parses the schema and attempts to ensure that we find an Analyze method for all types. Looks like graphTraversal was missing for example.

Reply:
Format is complete now, so not sure it's worth the time to stop. If we touch this file again, I'll do it.:)

---
In reply to: [305429964](https://github.com/microsoft/sarif-sdk/pull/1570#discussion_r305429964) [](ancestors = 305429964)